### DISCUSSION TITLE
how do I define a global vue function

### DISCUSSION THREAD

**Discussion by shi-yan** [quasarframework/quasar#12577] [General - Components / Directives / etc] [answered]
Hello, how can I define a global vue function that can be accessed by this.$... ?

examples I found https://stackoverflow.com/questions/63100658/add-global-variable-in-vue-js-3 requires touching the main.js file, I don't know how to achieve the same using quasar cli?

Thanks!

**Comment by MilosPaunovic** [comment, marked as answer]
What you need is a boot file. Check the exampe for `Axios` in the link below:
[https://quasar.dev/quasar-cli/boot-files#examples-of-boot-files](https://quasar.dev/quasar-cli/boot-files#examples-of-boot-files)
